src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,191 @@
+package duke.parser;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import duke.commands.AddCommand;
+import duke.commands.Command;
+import duke.commands.DeleteCommand;
+import duke.commands.ExitCommand;
+import duke.commands.FindCommand;
+import duke.commands.HelpCommand;
+import duke.commands.InvalidCommand;
+import duke.commands.ListCommand;
+import duke.commands.MarkCommand;
+import duke.commands.UnmarkCommand;
+import duke.exception.DukeException;
+
+/**
+ * The Parser class parses the user input.
+ *
+ * @author Chia Jeremy
+ */
+public class Parser {
+
+    private enum CommandCode {
+        TODO, DEADLINE, EVENT,
+        DELETE,
+        MARK, UNMARK,
+        FIND,
+        LIST,
+        HELP,
+        EXIT,
+        INVALID
+    }
+
+    /**
+     * Returns the command code of the user input.
+     *
+     * @param input the user input
+     * @return the command code of the input
+     */
+    public static CommandCode getCommand(String input) {
+        String command = input.split(" ", 2)[0].toLowerCase();
+        switch (command) {
+        case "todo":
+            return CommandCode.TODO;
+        case "deadline":
+            return CommandCode.DEADLINE;
+        case "event":
+            return CommandCode.EVENT;
+        case "delete":
+            return CommandCode.DELETE;
+        case "mark":
+            return CommandCode.MARK;
+        case "unmark":
+            return CommandCode.UNMARK;
+        case "find":
+            return CommandCode.FIND;
+        case "list":
+            return CommandCode.LIST;
+        case "help":
+            return CommandCode.HELP;
+        case "bye":
+            return CommandCode.EXIT;
+        default:
+            return CommandCode.INVALID;
+        }
+    }
+
+    /**
+     * Returns the command based on the user input.
+     *
+     * @param input the user input
+     * @return the command based on the input
+     */
+    public Command parseCommand(String input) {
```

Comment:
It seems that there's some form of inconsistency in this block of code. I think it might look neater if prepare() can change into like new TodoCommand() like the ListCommand(). What do you think?